app-i18n/fcitx-5.0.19: [31mFAILED: [0msrc/modules/spell/CMakeFiles/spell.dir/spell-custom-dict.cpp.o

```
[146/256] /usr/bin/loongarch64-unknown-linux-gnu-g++ -DFCITX_GETTEXT_DOMAIN=\"fcitx5\" -Dspell_EXPORTS -I/var/tmp/portage/app-i18n/fcitx-5.0.19/work/fcitx-5.0.19_build -I/var/tmp/portage/app-i18n/fcitx-5.0.19/work/fcitx5-5.0.19/src/lib/fcitx/.. -I/var/tmp/portage/app-i18n/fcitx-5.0.19/work/fcitx-5.0.19_build/src/lib/fcitx -I/var/tmp/portage/app-i18n/fcitx-5.0.19/work/fcitx5-5.0.19/src/lib/fcitx-config/.. -I/var/tmp/portage/app-i18n/fcitx-5.0.19/work/fcitx-5.0.19_build/src/lib/fcitx-config -I/var/tmp/portage/app-i18n/fcitx-5.0.19/work/fcitx5-5.0.19/src/lib/fcitx-utils/.. -I/var/tmp/portage/app-i18n/fcitx-5.0.19/work/fcitx-5.0.19_build/src/lib/fcitx-utils -isystem /usr/include/enchant-2 -isystem /usr/include/glib-2.0 -isystem /usr/lib64/glib-2.0/include  -Wall -Wextra -O2 -pipe -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -pthread -std=c++17 -MD -MT src/modules/spell/CMakeFiles/spell.dir/spell-custom-dict.cpp.o -MF src/modules/spell/CMakeFiles/spell.dir/spell-custom-dict.cpp.o.d -o src/modules/spell/CMakeFiles/spell.dir/spell-custom-dict.cpp.o -c /var/tmp/portage/app-i18n/fcitx-5.0.19/work/fcitx5-5.0.19/src/modules/spell/spell-custom-dict.cpp
[31mFAILED: [0msrc/modules/spell/CMakeFiles/spell.dir/spell-custom-dict.cpp.o 
/usr/bin/loongarch64-unknown-linux-gnu-g++ -DFCITX_GETTEXT_DOMAIN=\"fcitx5\" -Dspell_EXPORTS -I/var/tmp/portage/app-i18n/fcitx-5.0.19/work/fcitx-5.0.19_build -I/var/tmp/portage/app-i18n/fcitx-5.0.19/work/fcitx5-5.0.19/src/lib/fcitx/.. -I/var/tmp/portage/app-i18n/fcitx-5.0.19/work/fcitx-5.0.19_build/src/lib/fcitx -I/var/tmp/portage/app-i18n/fcitx-5.0.19/work/fcitx5-5.0.19/src/lib/fcitx-config/.. -I/var/tmp/portage/app-i18n/fcitx-5.0.19/work/fcitx-5.0.19_build/src/lib/fcitx-config -I/var/tmp/portage/app-i18n/fcitx-5.0.19/work/fcitx5-5.0.19/src/lib/fcitx-utils/.. -I/var/tmp/portage/app-i18n/fcitx-5.0.19/work/fcitx-5.0.19_build/src/lib/fcitx-utils -isystem /usr/include/enchant-2 -isystem /usr/include/glib-2.0 -isystem /usr/lib64/glib-2.0/include  -Wall -Wextra -O2 -pipe -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -pthread -std=c++17 -MD -MT src/modules/spell/CMakeFiles/spell.dir/spell-custom-dict.cpp.o -MF src/modules/spell/CMakeFiles/spell.dir/spell-custom-dict.cpp.o.d -o src/modules/spell/CMakeFiles/spell.dir/spell-custom-dict.cpp.o -c /var/tmp/portage/app-i18n/fcitx-5.0.19/work/fcitx5-5.0.19/src/modules/spell/spell-custom-dict.cpp
In file included from /var/tmp/portage/app-i18n/fcitx-5.0.19/work/fcitx5-5.0.19/src/modules/spell/spell-custom-dict.cpp:9:
/var/tmp/portage/app-i18n/fcitx-5.0.19/work/fcitx5-5.0.19/src/modules/spell/spell-custom-dict.h:34:17: error: ‘uint32_t’ was not declared in this scope

```


Full build log

[fcitx-5.0.19-full-build.log](https://github.com/microcai/gentoo-zh/files/9484143/fcitx-5.0.19-full-build.log)

--- a/app-i18n/fcitx/fcitx-5.0.19.ebuild
+++ b/app-i18n/fcitx/fcitx-5.0.19.ebuild
@@ -68,8 +68,11 @@ DEPEND="${RDEPEND}
 "
 
 src_prepare() {
-	pwd
 	ln -s "${DISTDIR}/fcitx-data-en_dict-20121020.tar.gz" src/modules/spell/en_dict-20121020.tar.gz || die
+
+	local PATCHES=(
+		"${FILESDIR}/fix-5.0.19-missing-includes-for-cstdint.patch"
+	)
 
 	cmake_src_prepare
 }

--- a/app-i18n/fcitx/files/fix-5.0.19-missing-includes-for-cstdint.patch
+++ b/app-i18n/fcitx/files/fix-5.0.19-missing-includes-for-cstdint.patch
@@ -1,0 +1,34 @@
+From d07db34b54b7ee512c086accaf3e73f0ba72abcc Mon Sep 17 00:00:00 2001
+From: Weng Xuetian <wengxt@gmail.com>
+Date: Sat, 3 Sep 2022 23:37:54 -0700
+Subject: [PATCH] Add missing includes for cstdint
+
+---
+ src/modules/spell/spell-custom-dict.cpp | 1 +
+ src/modules/spell/spell-custom-dict.h   | 1 +
+ 2 files changed, 2 insertions(+)
+
+diff --git a/src/modules/spell/spell-custom-dict.cpp b/src/modules/spell/spell-custom-dict.cpp
+index 867d352b..ae90d1d8 100644
+--- a/src/modules/spell/spell-custom-dict.cpp
++++ b/src/modules/spell/spell-custom-dict.cpp
+@@ -9,6 +9,7 @@
+ #include "spell-custom-dict.h"
+ #include <fcntl.h>
+ #include <sys/stat.h>
++#include <cstdint>
+ #include <stdexcept>
+ #include "fcitx-utils/cutf8.h"
+ #include "fcitx-utils/endian_p.h"
+diff --git a/src/modules/spell/spell-custom-dict.h b/src/modules/spell/spell-custom-dict.h
+index 1f5641ea..ad6cd6b8 100644
+--- a/src/modules/spell/spell-custom-dict.h
++++ b/src/modules/spell/spell-custom-dict.h
+@@ -8,6 +8,7 @@
+ #ifndef _FCITX_MODULES_SPELL_SPELL_CUSTOM_DICT_H_
+ #define _FCITX_MODULES_SPELL_SPELL_CUSTOM_DICT_H_
+
++#include <cstdint>
+ #include <string>
+ #include <vector>
+

